### DISCUSSION TITLE
Capturing RDL syntax errors

### DISCUSSION THREAD

**Discussion by hcadavid** [SystemRDL/systemrdl-compiler#106] [Q&A] [answered]
I'm integrating SystemRDL on a tool, and I'd like to capture RDL syntax error details (e.g. line number, mismatched symbol, etc).  The exception raised by RDLCompiler.compile_file doesn't capture such details (it just includes the message 'Parse aborted due to previous errors'). Is there some workaround to capture these details? they are indeed printed directly on STDERR when the error takes place!... Thanks.

**Comment by amykyta3** [comment, marked as answer]
Correct that by default, the compiler prints all messages to the `stderr` stream.
However if you need to, you can override the `MessagePrinter` class that is used when creating the `RDLCompiler` object.
See the `message_printer` argument to the compiler's constructor: https://systemrdl-compiler.readthedocs.io/en/latest/api/compiler.html#systemrdl.RDLCompiler

Extend the [MessagePrinter](https://systemrdl-compiler.readthedocs.io/en/latest/api/messages.html#systemrdl.messages.MessagePrinter) class and customize with your own implementation.

You can look at the default `MessagePrinter` implementation for more guidance on how it works: https://github.com/SystemRDL/systemrdl-compiler/blob/master/systemrdl/messages.py#L41
The `src_ref` object that is passed into the printer will provide file and line number context alongside the error message.

**Reply by amykyta3** [reply]
@hcadavid - I am planning to rework this mechanism and deprecate the MessageHandler/MessagePrinter in favor of Python's built-in logging.
Be sure to watch this issue for updates: https://github.com/SystemRDL/systemrdl-compiler/issues/168
